client/src/components/admin/AdminUsersListTable.tsx

```diff
@@ -0,0 +1,194 @@
+    }
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            users: props.users,
+            loadingUsers: {}
+        };
+    }
+
+    public render = () => {
+        const tableColNames: string[] = ["Name", "Email", "Phone", "Slack Username", "Have ID", "Admin", "Actions"];
+        const tableCols = tableColNames.map(col => (
+            <TableHeaderCell key={col} textAlign="center">{col}</TableHeaderCell>
+        ));
+
+        const tableRows = (changeAdmin: any) => this.state.users.map(user => (
+            <Table.Row key={user.uuid}>
+                <Table.Cell>{user.name}</Table.Cell>
+                <Table.Cell><a href={`mailto:${user.email}`}>{user.email}</a></Table.Cell>
+                <Table.Cell>{user.phone}</Table.Cell>
+                <Table.Cell>{user.slackUsername}</Table.Cell>
+                <Table.Cell>{AdminUsersListTable.checkOrX(user.haveID)}</Table.Cell>
+                <Table.Cell>{AdminUsersListTable.checkOrX(user.admin)}</Table.Cell>
+                <Table.Cell>{this.adminButton(user, changeAdmin)} <Button as={Link} primary basic compact size="tiny" to={`/user/${user.uuid}`}>Edit</Button></Table.Cell>
+            </Table.Row>
+        ));
+
+        const changeUserAdmin = gql`
+            mutation changeAdmin($uuid: String!, $admin:Boolean!) {
+                changeUserAdmin(uuid: $uuid, admin: $admin)
+            }
+        `;
+
+        return (
+            <Mutation mutation={changeUserAdmin}
+                      refetchQueries={[{ query: usersQuery }]}
+                      awaitRefetchQueries={true}
+            >
+                { (changeAdmin: any, { loading, data }: any) => (
+                    <Table compact celled textAlign="center">
+                        <Table.Header>
+                            <Table.Row>
+                                {tableCols}
+                            </Table.Row>
+                        </Table.Header>
+                        <Table.Body>
+                            {tableRows(changeAdmin)}
+                        </Table.Body>
+                    </Table>
+                )}
+            </Mutation>);
+    }
+
+    private getUserIndex(uuid: string) {
+        const users = this.state.users;
+        for (let i = 0; i < users.length; i++) {
+            if (users[i].uuid === uuid) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private modifyUserAdminInState = (uuid: string, admin: boolean) => {
+        const userIndex = this.getUserIndex(uuid);
+        const usersCopy = this.state.users;
+
+        const currentUser = usersCopy[userIndex];
+
+        currentUser.admin = admin;
+
+        usersCopy[userIndex] = currentUser;
+
+        this.setState({
+            users: usersCopy,
+        });
+    }
+
+    private modifyLoadingUsers = (uuid: string, loading: boolean) => {
+        const updatedLoadingUsers = this.state.loadingUsers;
```

Comment:
Regular users is an array since that's the format it comes in from the server and that's best for generating the table.  But working with a dictionary is much easier for keeping track of loading states since there's no iteration required, and it can start out empty because of the magical ` || false`.